pwnlib/tubes/process.py

```diff
@@ -345,33 +360,38 @@ def __init__(self, argv = None,
 
             p.success('pid %i' % self.pid)
 
-        if self.pty is not None:
-            if stdin is slave:
-                self.proc.stdin = os.fdopen(os.dup(master), 'r+b', 0)
-            if stdout is slave:
-                self.proc.stdout = os.fdopen(os.dup(master), 'r+b', 0)
-            if stderr is slave:
-                self.proc.stderr = os.fdopen(os.dup(master), 'r+b', 0)
-
-            os.close(master)
-            os.close(slave)
-
-        # Set in non-blocking mode so that a call to call recv(1000) will
-        # return as soon as a the first byte is available
-        if self.proc.stdout:
-            fd = self.proc.stdout.fileno()
-            fl = fcntl.fcntl(fd, fcntl.F_GETFL)
-            fcntl.fcntl(fd, fcntl.F_SETFL, fl | os.O_NONBLOCK)
-
-        # Save off information about whether the binary is setuid / setgid
-        self.suid = self.uid = os.getuid()
-        self.sgid = self.gid = os.getgid()
-        st = os.stat(self.executable)
-        if self._setuid:
-            if (st.st_mode & stat.S_ISUID):
-                self.suid = st.st_uid
-            if (st.st_mode & stat.S_ISGID):
-                self.sgid = st.st_gid
+        if sys.platform.startswith("linux"):
+            if self.pty is not None:
+                if stdin is slave:
+                    self.proc.stdin = os.fdopen(os.dup(master), 'r+b', 0)
+                if stdout is slave:
+                    self.proc.stdout = os.fdopen(os.dup(master), 'r+b', 0)
+                if stderr is slave:
+                    self.proc.stderr = os.fdopen(os.dup(master), 'r+b', 0)
+                os.close(master)
+                os.close(slave)
+        elif sys.platform.startswith("win"):
```

Comment:
... ???

Reply:
I removed it because ose.close(master) breaks on windows. Probably because there is no slave and master process on windows.